packages/main/src/backend/import/bankScraper.ts

```diff
@@ -23,20 +22,38 @@ export async function scrape(
   { companyId, credentials, startDate, timeout, showBrowser = false }: ScrapeParameters,
   emitProgressEvent: EmitProgressEventFunction,
   chromePath: string,
+  eventPublisher: EventPublisher,
 ) {
   const options: ScraperOptions = {
-    companyId, // mandatory; one of 'hapoalim', 'discount', 'otsarHahayal', 'leumiCard', 'isracard', 'amex'
-    startDate, // the date to fetch transactions from (can't be before the minimum allowed time difference for the scraper)
-    combineInstallments: false, // if set to true, all installment transactions will be combine into the first one
-    showBrowser, // shows the browser while scraping, good for debugging (default false)
-    verbose: false, // include more debug info about in the output
+    companyId,
+    startDate,
+    combineInstallments: false,
+    showBrowser,
+    verbose: false,
     executablePath: chromePath,
     defaultTimeout: timeout,
   };
+
   const scraper = createScraper(options);
-  scraper.onProgress((eventCompanyId: string, payload: { type: string }) => {
-    emitProgressEvent(companyId, payload.type);
-  });
-  const scrapeResult = await scraper.scrape(credentials);
-  return scrapeResult;
+  scraper.onProgress((eventCompanyId, payload) => emitProgressEvent(companyId, payload.type));
+
+  if (companyId === CompanyTypes.oneZero) {
+    const creds = credentials as typeof credentials & { otpLongTermToken: string; phoneNumber: string };
+    if (!creds.otpLongTermToken) {
+      await scraper.triggerTwoFactorAuth(creds.phoneNumber);
+      await eventPublisher.emit(EventNames.GET_OTP);
+      const otpCode = await new Promise<string>((resolve) => {
+        ipcMain.once('get-otp-response', (event, input) => resolve(input));
+      });
+      const result = await scraper.getLongTermTwoFactorToken(otpCode);
+      if ('longTermTwoFactorAuthToken' in result) {
+        await updateOtpLongTermToken(companyId, result.longTermTwoFactorAuthToken);
+        creds.otpLongTermToken = result.longTermTwoFactorAuthToken;
+      } else {
+        throw new Error('Failed to get long-term two-factor auth token');
+      }
```

Comment:
use `result.success`

```suggestion
      const result = await scraper.getLongTermTwoFactorToken(otpCode);
      if (result.success) {
        await updateOtpLongTermToken(companyId, result.longTermTwoFactorAuthToken);
        creds.otpLongTermToken = result.longTermTwoFactorAuthToken;
      } else {
        throw new Error('Failed to get long-term two-factor auth token');
      }
```